test/Riok.Mapperly.Tests/Mapping/UserMethodTest.cs

```diff
@@ -72,6 +72,68 @@ public void WithExistingInstance()
             .Be("target.StringValue = source.StringValue;");
     }
 
+    [Fact]
+    public void NullableToNonNullableShouldThrow()
+    {
+        var source = TestSourceBuilder.Mapping(
+            "A?",
+            "B",
+            "class A { }",
+            "class B { }");
+
+        TestHelper.GenerateSingleMapperMethodBody(source)
+            .Should()
+            .Be(@"if (source == null)
+        throw new System.ArgumentNullException(nameof(source));
+    var target = new B();
+    return target;".ReplaceLineEndings());
+    }
+
```

Comment:
I would also add tests for

- non-nullable to nullable
- nullable to nullable
- cases where the nullable context is disabled only for the source or the target?

Reply:
Added.